### PR TITLE
ceph.spec: Recommend (but do not require) podman

### PR DESCRIPTION
**Pull request by liewegas**
1- Strictly speaking, docker is okay too.
2- We don't want podman (or docker) installed inside the container image,
but we do need the cephadm package.

Fixes: https://tracker.ceph.com/issues/44065
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by dsavineau** [comment]
Do we need to update the deb packaging too even if this doesn't impact the container image ?

[1] https://github.com/ceph/ceph/blob/master/debian/control#L171

File: debian/control [hunk 1]
```diff
@@ -168,8 +168,8 @@ Description: debugging symbols for ceph-base
 
 Package: cephadm
 Architecture: linux-any
-Depends: docker.io,
-         lvm2,
+Recommends: podman | docker.io
+Depeends: lvm2,
```
Comment (dsavineau) [inline debian/control]: s/Depeends/Depends/

**Comment by liewegas** [comment]
```
error: line 398: Unknown tag: Recommends:     podman
```
I don't understand... there are other Recommends: lines in the spec file...

**Comment by liewegas** [comment]
> Unknown tag: Recommends

ah, i see the conditionals now

File: debian/control [hunk 2]
```diff
@@ -168,8 +168,8 @@ Description: debugging symbols for ceph-base
 
 Package: cephadm
 Architecture: linux-any
-Depends: docker.io,
-         lvm2,
+Recommends: podman | docker.io
```
Comment (tchaikov) [inline debian/control]: seems podman is not yet packaged for debian.

so i was reading 

- the issue at https://github.com/containers/libpod/issues/1742 tracking the progress of packaging libpod for debian, and 
- the ITP bug at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=930440, 
- https://ftp-master.debian.org/new/libpod_1.6.4+dfsg-1.html

so, two observations

- podman packaging is close
- the package name is podman =)

**Comment by tchaikov** [comment]
see also #33229, so we can have podman suggested on el8.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2020-02-12-1256/94ccf1c8c2f4724ca7c95d253cec438437252997/

**Pull request merged**